dropshot/src/config.rs

```diff
@@ -49,6 +49,32 @@ pub struct ConfigDropshot {
     pub bind_address: SocketAddr,
     /// maximum allowed size of a request body, defaults to 1024
     pub request_body_max_bytes: usize,
+    /// Default behavior for HTTP handler functions with respect to clients
+    /// disconnecting early.
+    pub default_handler_disposition: HandlerDisposition,
```

Comment:
```suggestion
    pub default_handler_disposition: HandlerDisposition,
```

Thinking out loud: how about: `on_client_disconnect` with the enum being called `OnClientDisconnect` with variants `CancelHandler` and `DetachHandler`?  I like being more concrete than the word "disposition".  On the other hand, it goes further down the confusing path of suggested we detach _when_ the client disconnects.

What about: `default_handler_task_mode` with `HandlerTaskMode` with variants `Detached` and `CancelOnDisconnect`?

Reply:
> Thinking out loud: how about: on_client_disconnect with the enum being called OnClientDisconnect with variants CancelHandler and DetachHandler? I like being more concrete than the word "disposition". On the other hand, it goes further down the confusing path of suggested we detach when the client disconnects.

This is almost word-for-word what I had in my first draft, and I scratched it for this exact reason.

> What about: default_handler_task_mode with HandlerTaskMode with variants Detached and CancelOnDisconnect?

That sounds good to me. :+1: